frontend/src/components/StakeSelectorItem.vue

```diff
@@ -82,136 +147,663 @@ export default {
+      case 'skill2':
+      case 'skill90':
+      case 'skill180':
+        return 'SKILL';
+      case 'king':
+      case 'king90':
+      case 'king180':
+        return 'KING';
+      case 'lp':
+      case 'lp2':
+        return 'SKILL-WBNB';
+      case 'cbkLandT1':
+      case 'cbkLandT2':
+      case 'cbkLandT3':
+        return 'CBKL';
+      default:
+        return 'unknown';
+      }
+    },
+
+    minimumStakeTimeFormatted() {
+      return formatDurationFromSeconds(this.stakeData.rewardMinimumStakeTime);
+    },
+
+    estimatedUnlockTimeLeftFormatted() {
+      return secondsToDDHHMMSS(this.stakeUnlockTimeLeftCurrentEstimate);
+    },
+
+    showRewardClaimSection() {
+      if (this.rewardClaimState === 'loading') {
+        return true;
+      }
+
+      return toBN(this.currentRewardEarned).gt(0);
+    },
+
+    stakedBalance() {
+      return toBN(this.stakeData.stakedBalance);
+    },
+
+    remainingCapacityForDeposit() {
+      if (!this.stakeData.remainingCapacityForDeposit) {
+        return null;
+      }
+
+      return toBN(this.stakeData.remainingCapacityForDeposit);
+    },
+
+    remainingCapacityForWithdraw() {
+      return toBN(this.stakeData.remainingCapacityForWithdraw);
+    },
+
+    contractBalance() {
+      return toBN(this.stakeData.contractBalance);
+    },
+
+    currentState() {
+      if (!this.defaultAccount || this.errorWhenUpdating) {
+        return 'connectWallet';
+      }
+
+      if (!this.isDeposit && this.unlockTimeLeftInternal > 0) {
+        return 'stakeLocked';
+      }
+
+      if (
+        this.remainingCapacityForDeposit &&
+        this.remainingCapacityForDeposit.eq(0) &&
+        this.isDeposit
+      ) {
+        return 'contractFull';
+      }
+
+      if (
+        this.remainingCapacityForDeposit &&
+        this.bigNumberAmount.gt(this.remainingCapacityForDeposit) &&
+        this.isDeposit
+      ) {
+        return 'amountIsTooBig';
+      }
+
+      if (this.textAmount[this.textAmount.length - 1] === '.') {
+        return 'waiting';
+      }
+
+      if (isStakeType(this.stakeType) && this.textAmount <= 0) {
+        return 'inputIsZero';
+      }
+
+      if (isNftStakeType(this.stakeType) && this.idsToStake.length === 0) {
+        return 'inputIsZero';
+      }
+
+      const hasSufficientBalance = this.isDeposit
+        ? toBN(this.walletBalance).gte(this.bigNumberAmount)
+        : toBN(this.stakedBalance).gte(this.bigNumberAmount);
+
+      if (!hasSufficientBalance) {
+        return 'insufficientBalance';
+      }
 
-      if (this.rewardDistributionTimeLeft > 0) {
-        exclamationMark += 'green-exclamation-mark ';
+      if (this.bigNumberAmount.gt(this.contractBalance) && !this.isDeposit) {
+        return 'notEnoughFundsInExitPool';
+      }
+
+      if (
+        this.bigNumberAmount.gt(this.remainingCapacityForWithdraw) &&
+        !this.isDeposit
+      ) {
+        return 'notEnoughFundsInExitPool';
+      }
+
+      return 'ok';
+    },
```

Comment:
Nasty function, hard to follow. Would appreciate some clean-up splitting it out or some commenting to explain what it is doing for future readers.

Reply:
I will. This is a function I just copied from the old staking component.